js/views/modals/Settings/Page.js

```diff
@@ -37,27 +92,46 @@ export default class extends baseVw {
     this.profile.set(formData);
 
     const save = this.profile.save();
+    let saveAvatar;
+    let saveHeader;
 
     this.trigger('saving');
 
     if (!save) {
       // client side validation failed
       this.trigger('saveComplete', true);
+      this.render();
```

Comment:
The render needs to happen after any uploads of the avatar and header, otherwise the new images are replaced by the old ones.